src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  function add(a, b) {
+    return a + b;
+  }
+
+  function subtract(a, b) {
+    return a - b;
+  }
+
+  function multiply(a, b) {
+    return a * b;
+  }
+
+  function divide(a, b) {
+    return a / b;
+  }
+
+  function operate(callback, num) {
+    result = callback(result, num);
+
+    return this;
+  }
+
+  function reset() {
+    result = 0;
+
+    return this;
+  }
+
+  return {
+    get result() {
+      return result;
```

Comment:
The getter for `result` is unnecessary since `result` is already private to the `makeCalculator` function scope. You can directly return the `result` property in the object.